src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -420,4 +420,14 @@ public function testVeryLongQuotedStrings()
 
         $this->assertEquals($longStringWithQuotes, $arrayFromYaml['longStringWithQuotes']);
     }
+
+    public function testBooleanMappingKeysAreConvertedToStrings()
+    {
+        $this->assertSame(array('false' => 'foo'), Inline::parse('{false: foo}'));
```

Comment:
btw, if we start evaluating, we would also have to forbid floats to avoid breaking BC.